l10n_nl_tax_invoice_basis/models/account_tax.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    def get_move_line_partial_domain(self, from_date, to_date, company_id):
+        company = self.env['res.company'].browse(company_id)
+        is_nl = company.country_id == self.env.ref('base.nl')
+        if is_nl:
+            return [
+                ('company_id', '=', company_id),
+                '|',
+                '&',
+                ('tax_date', '!=', False),
+                '&',
+                ('tax_date', '<=', to_date),
+                ('tax_date', '>=', from_date),
+                '&',
+                ('tax_date', '=', False),
+                '&',
+                ('date', '<=', to_date),
+                ('date', '>=', from_date),
+            ]
```

Comment:
Though this domain list works properly, it is rather hard to read and surely not understandable at a first glance.
Can I suggest you to split it in order to be more readable? Something like this could already be more readable:

```
tax_date_condition = ['&', '&', ('tax_date', '!=', False), ('tax_date', '<=', to_date), ('tax_date', '>=', from_date)]
date_condition = ['&', '&', ('tax_date', '=', False), ('date', '<=', to_date), ('date', '>=', from_date)]
return [('company_id', '=', company_id), '|',] + tax_date_condition + date_condition
```

In addition, it could also be useful to add a comment to explain what the condition means, either in natural language or at least using a much more readable infix notation.